### ISSUE TITLE
Compatibility with "Sophisticated wolves"

### ISSUE THREAD

**Issue by raspopov** [CenturionFox/wolfarmor#9]
Is it possible to make this mod compatible with "[Sophisticated wolves](https://mods.curse.com/mc-mods/minecraft/220495-sophisticated-wolves)" mod? I'd like to have both, please...

**Comment by piks** [comment]
Yes, there is the alpha for 1.10 but there is no alpha build with sophisticated wolves for 1.11.2 from what I see, unless the release supports it

**Comment by CenturionFox** [comment]
Hey,

The answer is:
For versions 1.10+: Yes!  I've been slow in updating the mod to 1.11 due to time constraints, but plan to release the 1.11 version (and maybe the 1.12 branch) by early July.
For version 1.7: Ehhhhhhhhhh......

The problem with 1.7 is that it is from before forge added the capability system, which is the definitive linchpin in gsbaoge's fix (Issue #4).  Therefore, I have to implement some other method of compatibility within the 1.7 version.  These include:
- Replacing EntityWolf.class at load time with a modified version (which is definitively **not allowed** since I'd be replacing the entire class, and therefore distributing Minecraft code within my mod)
- Writing custom methods in Java bytecode and patching them in through ASM (which I've worked with before, but all I did then was add some custom enchantments and I scrapped it since my helper classes were WAY too complex)
- Somehow patching the EntityWolf bytecode with a precompiled binary diff file (which I uh... I'm not sure how to go about that one.  That would probably require a ridiculous number of classloader customizations, which I have never even attempted before)

TL;DR:  Yes and "maybe?"; 1.10+ is a definite "Yep", while 1.7 would require some weird workarounds that I'm not too sure about implementing.
